onisite/settings_base.py

```diff
@@ -1,74 +1,215 @@
+_ONEWEEK = _ONEDAY * 7
+
+
+####################################################################
+# DJANGO CUSTOMIZATIONS
+####################################################################
+# Hosts/domain names valid for this site
+# Should be overridden in settings_local.py, especially for production
+# TODO: Enable overriding by /etc/openoni.ini
+ALLOWED_HOSTS = ['*']
+
+
+CACHES = {
+    'default': {
+        # Development
+        'BACKEND': 'django.core.cache.backends.dummy.DummyCache'
+
+        # Production
+        #'BACKEND': 'django.core.cache.backends.filebased.FileBasedCache',
+        #'LOCATION': '/var/tmp/django_cache',
+        #'TIMEOUT': _ONEWEEK * 8
+    }
+}
+
+
+# Overridden in settings_local.py or /etc/openoni.ini
 DATABASES = {
     'default': {
         'ENGINE': 'django.db.backends.mysql',
         'HOST': 'rdbms',
         'NAME': 'openoni',
         'USER': 'openoni',
         'PASSWORD': 'openoni',
-        }
     }
+}
 
-# Make this unique, and don't share it with anybody.  This MUST be overridden
-# either in settings_local.py or /etc/openoni.ini.
-SECRET_KEY = ''
+# List of configuration classes / app packages in order of priority (i.e., the
+# first item in the list has final say when collisions occur)
+INSTALLED_APPS = (
+    # Default
+#    'django.contrib.admin',
+#    'django.contrib.auth',
+#    'django.contrib.contenttypes',
+    'django.contrib.sessions',
+    'django.contrib.messages',
+    'django.contrib.staticfiles',
 
-# Classes which implement request/response middleware
+    # OpenONI
+    'django.contrib.humanize',  # Added to make data more human-readable
+    'themes.default',
+    'core',
+)
+
+# Suggested order: https://docs.djangoproject.com/en/1.9/ref/middleware/#middleware-ordering
 MIDDLEWARE_CLASSES = (
+    'django.middleware.security.SecurityMiddleware',
+    'core.middleware.TooBusyMiddleware',                               # OpenONI
+    'django.middleware.http.ConditionalGetMiddleware',                 # OpenONI
+    'django.contrib.sessions.middleware.SessionMiddleware',
     'django.middleware.common.CommonMiddleware',
     'django.middleware.csrf.CsrfViewMiddleware',
-    'django.middleware.http.ConditionalGetMiddleware',
-    'core.middleware.TooBusyMiddleware',
+#    'django.contrib.auth.middleware.AuthenticationMiddleware',
+#    'django.contrib.auth.middleware.SessionAuthenticationMiddleware',
+    'django.contrib.messages.middleware.MessageMiddleware',
+    'django.middleware.clickjacking.XFrameOptionsMiddleware',
 )
 
+# Site id differentiates DB data if multiple sites use the same database
+SITE_ID = 1
+
+# Use cache with storing static files; e.g., CSS, images, etc.
+# Development (Default)
+#STATICFILES_STORAGE = 'django.contrib.staticfiles.storage.StaticFilesStorage'
+# Production
+#STATICFILES_STORAGE = 'django.contrib.staticfiles.storage.CachedStaticFilesStorage'
+
+# URL prepended to static file paths; Must include trailing slash
+STATIC_URL = '/media/'
```

Comment:
This could be bad if people already using ONI (say, UO) don't get around to changing their local configs.  I don't want all our static media URLs to suddenly break for people who have cached the site.  For whatever reason, many pages are cached for a long time, and browsers will keep those cached and request the old URL for assets.

That being said, it wouldn't be the end of the world as long as we make it clear that things may break when upgrading to this version of ONI.  And we give it a major version bump (i.e., 2.0.0 instead of the next 1.x)

Reply:
Yeah, I think we're in major version bump territory with this process.

This would only break links if people are linking to the site's images for presentation (like a header background or sponsor logo images in the footer), not to newspaper pages etc. The URLs as served by Django would all be updated and served and cached anew if we change the path.

If anyone already deployed is worried, I was hoping to keep a list of the defaults we change and instructions on what must be set in local settings to override and retain the current (soon-to-be old) settings. I imagine there might be other migration steps we'll need to document, though nothing I've seen so far with 1.9.

Reply:
So, now I see what `/static/` is currently being used for. I guess I didn't look very closely earlier. In the [Apache config](https://github.com/open-oni/open-oni/blob/master/conf/openoni.conf#L38) it's used for `robots.txt` and `sitemap.xml`. Only the robots.txt file is covered by the repo. In my local files not tracked by git, the `coverage` library has created some static HTML pages about code coverage. This must be done during the initial install?

Having `robots.txt` and `sitemap.xml` not at the root of the domain is fairly uncommon, so I think that may be another reason to change things here.

For `STATIC_URL`, I'm still leaning towards `/static/` or maybe even just at `/` so they're served from `https://(hostname)/js/` instead of a sub-URI. The latter is the Django default, from how I understand the documentation. It also depends on whether or not we want to apply Apache directives to the static files, as is currently done where Apache adds caching headers. I think it'd be simpler to handle all of that inside Django though.

Reply:
Letting Apache serve static files may be a bit more complicated, but it is a *huge* performance gain.  Django is simply not a fast platform when it has to do the routing of everything there is.

When we changed it to make Apache serve the word coordinate files, the difference was dramatic - faster page loads and less CPU pegging by django.

(Feel free to replace "Apache" with any static asset server you feel like propping up in front of django - point is that django is simply not optimized for this kind of thing like Apache/nginx/etc)

Reply:
I'm going to +1 the idea of having a specific static files directory. 

We have been leveraging /media for handling the marc files that need to be locally read by ONI (for titles that require customized marc.xml that isn't available from LC/OCLC). 

I agree if we're going to change this, it is better A) sooner than later; B) that it be very clearly labeled as part of the transition so folks know that they need to change their configs.

Reply:
Okay, I think we should go with `/static/` then. I'll change that now and will configure Apache to serve the static files later when working on Docker deployment etc.